### ISSUE TITLE
API is missing an EN_setcurvetype function

### ISSUE THREAD

**Issue by LRossman** [OpenWaterAnalytics/EPANET#575]
The v2.2 toolkit API has an `EN_getcurvetype` function that retrieves the type of a curve (`PUMP`, `VOLUME`. etc.) but no complementary `EN_setcurvetype` function that sets a curve's type. Such a function would be useful when using the API to build a network database. For example, one would like to designate a newly created curve as a `PUMP `curve before assigning it to a Pump object. Right now all newly created curves are designated as `GENERIC `curves and there is no way to change this.

Here's the code to implement such a function:
```
int DLLEXPORT EN_setcurvetype(EN_Project p, int curveIndex, int curveType)
{
    Network *net = &p->network;
    if (!p->Openflag) return 102;
    if (curveIndex < 1 || curveIndex > net->Ncurves) return 206;
    if (curveType < 0 || curveType > EN_GENERIC_CURVE) return 251;  
    net->Curve[curveIndex].Type = curveType;
    return 0;
}
```

**Comment by eladsal** [comment]
Hi @LRossman,

When I added the `ENgetcurvetype` I remember considering the `set` function and thought it was not needed. The idea was to set the curve type according to its use. If a curve is set to a pump then it is a pump curve, if it is set to a tank than it is a volume curve and so on. Since your suggestion does not remove these settings why do you need the `EN_setcurvetype` function? Maybe it is for the calling application use? On the other hand, I don't really mind adding the function as it does not make any difference with your current code suggestion.

**Comment by LRossman** [comment]
In the current code curve types are only assigned when an input file is read and when a pump is assigned a head curve using `EN_setlinkvalue` or `EN_setheadcurveindex`. If a new curve is created using `EN_addcurve` its type always stays as` GENERIC` (unless it's assigned to a pump) even though it may be used as a `VOLUME` or `EFFICIENCY` curve later on.

I admit that this is a very minor issue since a curve's designated type doesn't affect the hydraulic calculations - only it's data points matter. However, while working on my LEPANET GUI I wanted to tell the visual curve editor what type of newly created curve was being edited, so it could label its data columns correctly and display the correct curve shape in a plot. That's when I discovered there was no way to use the toolkit API to do this, hence my proposed new function.

In addition to adding EN_setcurvetype, we should also have the toolkit internally modify a curve's type whenever it is assigned as a VOLUME, EFFICIENCY, or HEAD_LOSS curve (but see issue #576 regarding the latter curve type).

**Comment by eladsal** [comment]
I thought it had to do with the calling program (LEPANET). This addition is fine by me.

**Comment by LRossman** [comment]
This issue has been successfully addressed in PR #580.
